pkgs/by-name/zl/zlibrary-desktop/package.nix

```
@@ -0,0 +1,201 @@
+{
+  lib,
+  fetchzip,
+  stdenvNoCC,
+
+  autoPatchelfHook,
+  makeWrapper,
+  copyDesktopItems,
+  makeDesktopItem,
+  undmg,
+
+  nss,
+  nspr,
+  at-spi2-atk,
+  cups,
+  dbus,
+  libdrm,
+  gtk3,
+  xorg,
+  libgbm,
+  libxkbcommon,
+  alsa-lib,
+  libGL,
+  glibc,
+  pango,
+  cairo,
+  expat,
+  libgcc,
+  libffi,
+  glib,
+  zlib,
+  pcre2,
+  util-linux,
+  libselinux,
+  avahi,
+  gnutls,
+  harfbuzz,
+  fontconfig,
+  fribidi,
+  gdk-pixbuf,
+  libepoxy,
+  tinysparql,
+  libthai,
+  libpng,
+  pixman,
+  p11-kit,
+  libidn2,
+  libunistring,
+  libtasn1,
+  nettle,
+  gmp,
+  libcap,
+  wayland,
+  graphite2,
+  bzip2,
+  brotli,
+  libjpeg,
+  json-glib,
+  libxml2,
+  sqlite,
+  libdatrie,
+  vulkan-loader,
+  mesa,
+  gvfs,
+  libudev-zero,
+  pciutils,
+  libGLX,
+  zstd,
+}:
+
+let
+  pname = "zlibrary-desktop";
+  version = "2.4.1";
+
+  meta = {
+    homepage = "https://z-library.sk";
+    description = "client for the online library Z-Library";
+    license = lib.licenses.unfree;
+    platforms = [ "x86_64-linux" ];
+    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
+    maintainers = with lib.maintainers; [ ulysseszhan ];
+    mainProgram = "z-library";
+  };
+
+  stdenv = stdenvNoCC;
+
+  libraries =
+    [
+      nss
+      nspr
+      at-spi2-atk
+      cups
+      dbus
+      libdrm
+      gtk3
+      libgbm
+      libxkbcommon
+      alsa-lib
+      libGL
+      glibc
+      expat
+      libgcc
+      libffi
+      glib
+      zlib
+      pcre2
+      util-linux
+      libselinux
+      avahi
+      gnutls
+      harfbuzz
+      fontconfig
+      fribidi
+      gdk-pixbuf
+      libepoxy
+      tinysparql
+      libthai
+      libpng
+      pixman
+      p11-kit
+      libidn2
+      libunistring
+      libtasn1
+      nettle
+      gmp
+      libcap
+      wayland
+      graphite2
+      bzip2
+      brotli
+      libjpeg
+      json-glib
+      libxml2
+      sqlite
+      libdatrie
+      vulkan-loader
+      mesa
+      gvfs
+      libudev-zero
+      pciutils
+      libGLX
+      zstd
+    ]
+    ++ (with xorg; [
+      libXcomposite
+      libXdamage
+      libXfixes
+      libXrandr
+      libXext
+      libX11
+      libxcb
+      libXi
+      libXrender
+      libXau
+      libXdmcp
+      libXcursor
+      libXinerama
+      libxshmfence
+    ]);
+in
+stdenv.mkDerivation {
+  inherit pname version meta;
+
+  src = fetchzip {
+    url = "https://web.archive.org/web/20250401040353/https://s3proxy.cdn-zlib.sk/te_public_files/soft/linux/zlibrary-setup-latest.tar.gz";
+    hash = "sha256-nL/pWTmmiPacu81xfxU6UwVqtVinWGUzN00FWCvZEfw=";
+  };
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    makeWrapper
+    copyDesktopItems
+  ] ++ libraries;
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = pname;
+      desktopName = "Z-Library";
+      comment = meta.description;
```

these are both static strings that are unlikely to change, see https://github.com/NixOS/nixpkgs/issues/277994
```suggestion
      name = "zlibrary-desktop";
      desktopName = "Z-Library";
      comment = "client for the online library Z-Library;
```